firedrake/functionspacedata.py

```diff
@@ -538,13 +540,28 @@ def get_map(self, V, entity_set, map_arity, bcs, name, offset, parent):
             else:
                 new_entity_node_list = entity_node_list
 
+            if kind == "interior_facet" and self.bt_masks is not None:
+                bt_masks = {}
+                off = map_arity // 2
+                for method, (bottom, top) in iteritems(self.bt_masks):
+                    b = []
+                    t = []
+                    for i in bottom:
+                        b.append(i)
+                        b.append(i+off)
+                    for i in top:
+                        t.append(i)
+                        t.append(i+off)
+                    bt_masks[method] = (b, t)
```

Comment:
I'm not sure I understand this. Surely, top and bottom BCs have no effect on most interior horizontal facet integrals, except for the lowest interior horizontal facet layer, where bottom masks are to be applied for the bottom cell, and for the highest interior horizontal facet layer, where top masks are to applied for the top cell. But here in the mask you seem to apply both top and bottom masks for both cells. What's going on here?